src/runner.rs

```diff
@@ -91,7 +91,7 @@ impl<Context: Clone + Send + Sync + 'static> Runner<Context> {
                     shutdown_when_queue_empty: self.shutdown_when_queue_empty,
                     poll_interval: queue.poll_interval,
                     jitter: queue.jitter,
-                    archive_completed_jobs: queue.archive_completed_jobs,
+                    archive_completed_jobs: queue.archive_completed_jobs.clone(),
```

Comment:
Why is that clone needed here?

Reply:
Ah, I overlooked the loop (which was outside that immediate change). The reason is that a `bool` is `Copy`, while your complex enum type is not. You can't derive `Copy`, because of the `Predicate` variant, which contains a function, which is also not `Copy`. So, all good. Apologies for not seeing that earlier.